authenticatorbridge/src/test/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerTest.kt

```diff
@@ -0,0 +1,440 @@
+        every { mockBridgeService.registerBridgeServiceCallback(capture(callback)) } just runs
+        every {
+            anyConstructed<Intent>().setComponent(any())
+        } returns mockIntent
+        every { context.unbindService(any()) } just runs
+        every {
+            context.bindService(
+                any(),
+                capture(serviceConnection),
+                Context.BIND_AUTO_CREATE
+            )
+        } returns true
+
+        fakeLifecycleOwner.lifecycle.dispatchOnStart()
+        serviceConnection.captured.onServiceConnected(mockk(), mockk())
+        assertEquals(AccountSyncState.Loading, manager.accountSyncStateFlow.value)
+
+        fakeSymmetricKeyStorageProvider.symmetricKey = null
+        callback.captured.onAccountsSync(encryptedAccounts)
+        assertEquals(AccountSyncState.Error, manager.accountSyncStateFlow.value)
+
+        verify(exactly = 0) { mockBridgeService.symmetricEncryptionKeyData }
+        verify { context.bindService(any(), any(), Context.BIND_AUTO_CREATE) }
+        verify { mockBridgeService.registerBridgeServiceCallback(any()) }
+        verify { mockBridgeService.syncAccounts() }
+    }
+
+    @Test
+    @Suppress("MaxLineLength")
+    fun `onAccountsSync when decryption fails state should be error`() {
+        val encryptedAccounts: EncryptedSharedAccountData = mockk()
+        mockkStatic(EncryptedSharedAccountData::decrypt)
+        every { encryptedAccounts.decrypt(SYMMETRIC_KEY) } returns Result.failure(RuntimeException())
+        every {
+            mockBridgeService.checkSymmetricEncryptionKeyFingerprint(
+                SYMMETRIC_KEY.toFingerprint().getOrNull()
+            )
+        } returns true
+        fakeSymmetricKeyStorageProvider.symmetricKey = SYMMETRIC_KEY
+        val serviceConnection = slot<ServiceConnection>()
+        val callback = slot<IAuthenticatorBridgeServiceCallback>()
+        val mockIntent: Intent = mockk()
+        every { IAuthenticatorBridgeService.Stub.asInterface(any()) } returns mockBridgeService
+        every { mockBridgeService.registerBridgeServiceCallback(capture(callback)) } just runs
+        every {
+            anyConstructed<Intent>().setComponent(any())
+        } returns mockIntent
+        every { context.unbindService(any()) } just runs
+        every {
+            context.bindService(
+                any(),
+                capture(serviceConnection),
+                Context.BIND_AUTO_CREATE
+            )
+        } returns true
+
+        fakeLifecycleOwner.lifecycle.dispatchOnStart()
+        serviceConnection.captured.onServiceConnected(mockk(), mockk())
+        assertEquals(AccountSyncState.Loading, manager.accountSyncStateFlow.value)
+
+        callback.captured.onAccountsSync(encryptedAccounts)
+        assertEquals(AccountSyncState.Error, manager.accountSyncStateFlow.value)
+
+        assertEquals(fakeSymmetricKeyStorageProvider.symmetricKey, SYMMETRIC_KEY)
```

Comment:
I think the order of this assertion is backwards right? `SYMMETRIC_KEY` is the expected value, so it should be first.

Reply:
Ah, yep, those are backwards: https://github.com/bitwarden/android/commit/08264b27d598a86435238a14587e34d90d9895a6

Reply:
There are still some of these that are mixed up.